storage/src/test/java/org/apache/kafka/server/log/remote/storage/RemoteLogManagerConfigTest.java

```diff
@@ -16,184 +16,97 @@
+
         // Removing remote.log.metadata.manager.class.name so that the default value gets picked up.
-        if (useDefaultRemoteLogMetadataManagerClass) {
-            props.remove(REMOTE_LOG_METADATA_MANAGER_CLASS_NAME_PROP);
-        }
-        TestConfig config = new TestConfig(props);
-        RemoteLogManagerConfig remoteLogManagerConfig = new RemoteLogManagerConfig(config);
-        assertEquals(expectedRemoteLogManagerConfig, remoteLogManagerConfig);
-    }
+        props.remove(REMOTE_LOG_METADATA_MANAGER_CLASS_NAME_PROP);
+
+        RemoteLogManagerConfig remoteLogManagerConfig = new RemoteLogManagerConfig(props);
+        assertEquals(expectedRemoteLogManagerConfig.values(), remoteLogManagerConfig.values());
+
+        assertEquals(rsmProps, remoteLogManagerConfig.remoteStorageManagerProps());
+        assertEquals(rlmmProps, remoteLogManagerConfig.remoteLogMetadataManagerProps());
 
-    private static RemoteLogManagerConfig getRemoteLogManagerConfig(boolean useDefaultRemoteLogMetadataManagerClass,
-                                                                    String rsmPrefix,
-                                                                    String rlmmPrefix,
-                                                                    Map<String, Object> rsmProps,
-                                                                    Map<String, Object> rlmmProps) {
-        String remoteLogMetadataManagerClass = useDefaultRemoteLogMetadataManagerClass ? DEFAULT_REMOTE_LOG_METADATA_MANAGER_CLASS_NAME : "dummy.remote.log.metadata.class";
-        return new RemoteLogManagerConfig(true,
-                "dummy.remote.storage.class",
-                "dummy.remote.storage.class.path",
-                remoteLogMetadataManagerClass,
-                "dummy.remote.log.metadata.class.path",
-                "listener.name",
-                1024 * 1024L,
-                1,
-                1,
-                1,
-                60000L,
-                100L,
-                60000L,
-                0.3,
-                10,
-                100,
-                100,
-                rsmPrefix,
-                rsmProps,
-                rlmmPrefix,
-                rlmmProps,
-                Long.MAX_VALUE,
-                11,
-                1,
-                Long.MAX_VALUE,
-                11,
-                1,
-                500);
+        // Even with empty properties, RemoteLogManagerConfig has default values
+        Map<String, Object> emptyProps = new HashMap<>();
+        RemoteLogManagerConfig remoteLogManagerConfigEmptyConfig = new RemoteLogManagerConfig(emptyProps);
+        assertEquals(remoteLogManagerConfigEmptyConfig.values().size(), 28);
+
+        // Test with a empty string props should throw ConfigException
+        Map<String, Object> emptyStringProps = new HashMap<>();
+        emptyStringProps.put(REMOTE_LOG_METADATA_MANAGER_LISTENER_NAME_PROP, "");
+        assertThrows(ConfigException.class, () ->
```

Comment:
This assertion doesn't fit the name of the test. Split these three different tests apart, they have nothing to do with one another.

Reply:
Yea, it's better to split them. Done.